.gitignore

```
@@ -17,6 +17,9 @@ __pycache__/
 .ruff_cache/
 .tox/
 
+#python virtual environement
+taskgraph/
```

Nit: Let's make git ignore just the `taskgraph` folder at the root directory. The current implementation excludes any files in `src/taskgraph` which will eventually be an issue. This can easily be fixed by:

```suggestion
/taskgraph/
```

Source[1]: 

> The pattern `hello.*` matches any file or directory whose name begins with `hello.`. If one wants to restrict this only to the directory and not in its subdirectories, one can prepend the pattern with a slash, i.e. `/hello.*`; the pattern now matches `hello.txt`, `hello.c` but not `a/hello.java`.


[1] https://git-scm.com/docs/gitignore#_examples